### ISSUE TITLE
Candeo C202 Rotary Dimmer: Unable to Set Brightness

### ISSUE THREAD

**Issue by KingJ** [Koenkk/zigbee-herdsman-converters#2968]
I recently picked up a [Candeo C202 Zigbee Rotary Dimmer](https://candeo.io/store/zigbee-dimmer-rotary-switch-c202/), which zigbee2mqtt identified as an "EcoDim Eco-Dim.07". On/off and brightness reporting/setting worked without issue, and so I picked up a second one. However, this time it was not recognised by zigbee2mqtt - and a model of "HK-DIM-A" is reported instead. The documentation included is ever so slightly different (minor text corrections etc), but the dimmer itself looks identical to before so I can only assume that the internals have changed.

As the device was not recognised, I've tried to add it as a new device [following the instructions here](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html#how-to-support-new-devices).

My current, fairly basic, external converter file for the device is as follows;

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['HK-DIM-A'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'HK-DIM-A', // Vendor model number, look on the device for a model number
    vendor: 'Candeo', // Vendor of the device (only used for documentation and startup logging)
    description: 'Zigbee LED Dimmer Smart Switch', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    extend: extend.light_onoff_brightness({noConfigure: true, disableEffect: true}),
};

module.exports = definition;
```

With this in place, I am able to report on and control the on/off state, but only report on the brightness. Attempting to set the brightness via zigbee2mqtt results in no change in brightness. As soon as I physically adjust the dimmer, the reported brightness value snaps back to the real value (i.e. if it was at 50, then it I set it in zigbee2mqtt to 100, adjusting the dimmer physically makes it jump to 56). I've also tried copying the configure block used for the [Eco-Dim.07](https://github.com/Koenkk/zigbee-herdsman-converters/blob/c1f440e3a63948266290eaacf6da779c026b1806/devices/ecodim.js#L41-L52) and [HK-LN-DIM-A](https://github.com/Koenkk/zigbee-herdsman-converters/blob/c1f440e3a63948266290eaacf6da779c026b1806/devices/led_trading.js#L11-L16) devices given the original hardware for the former, and the similarity in name for the latter, but to no avail.

I'm still quite new to Zigbee so i'm not quite sure how to troubleshoot this further - any pointers would be greatly appreciated! From reading other bug reports, a snippet from the device database has been useful and so here's the database entry for the `HK-DIM-A` unrecognised device;

```
{
    "id": 4,
    "type": "Router",
    "ieeeAddr": "0x000d6ffffec923fd",
    "nwkAddr": 18607,
    "manufId": 4107,
    "manufName": "Candeo",
    "powerSource": "Mains (single phase)",
    "modelId": "HK-DIM-A",
    "epList": [
        1,
        242
    ],
    "endpoints": {
        "1": {
            "profId": 260,
            "epId": 1,
            "devId": 257,
            "inClusterList": [
                0,
                3,
                4,
                5,
                6,
                8,
                2821,
                4096
            ],
            "outClusterList": [
                25
            ],
            "clusters": {
                "genBasic": {
                    "attributes": {
                        "modelId": "HK-DIM-A",
                        "manufacturerName": "Candeo",
                        "powerSource": 1,
                        "zclVersion": 3,
                        "appVersion": 3,
                        "stackVersion": 6,
                        "hwVersion": 1,
                        "dateCode": "20210128",
                        "swBuildId": "1.00"
                    }
                },
                "genOnOff": {
                    "attributes": {
                        "onOff": 1
                    }
                },
                "genLevelCtrl": {
                    "attributes": {
                        "currentLevel": 201
                    }
                },
                "genScenes": {
                    "attributes": {
                        "currentScene": 0
                    }
                },
                "genIdentify": {
                    "attributes": {
                        "identifyTime": 0
                    }
                },
                "lightingBallastCfg": {
                    "attributes": {}
                }
            },
            "binds": [
                {
                    "cluster": 8,
                    "type": "endpoint",
                    "deviceIeeeAddress": "0x00124b0022a51f6b",
                    "endpointID": 1
                },
                {
                    "cluster": 6,
                    "type": "endpoint",
                    "deviceIeeeAddress": "0x00124b0022a51f6b",
                    "endpointID": 1
                }
            ],
            "configuredReportings": [
                {
                    "cluster": 6,
                    "attrId": 0,
                    "minRepIntval": 0,
                    "maxRepIntval": 3600,
                    "repChange": 0
                },
                {
                    "cluster": 8,
                    "attrId": 0,
                    "minRepIntval": 0,
                    "maxRepIntval": 3600,
                    "repChange": 1
                }
            ],
            "meta": {}
        },
        "242": {
            "profId": 41440,
            "epId": 242,
            "devId": 97,
            "inClusterList": [],
            "outClusterList": [
                33
            ],
            "clusters": {},
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        }
    },
    "appVersion": 3,
    "stackVersion": 6,
    "hwVersion": 1,
    "dateCode": "20210128",
    "swBuildId": "1.00",
    "zclVersion": 3,
    "interviewCompleted": true,
    "meta": {
        "configured": 1461352984
    },
    "lastSeen": 1630183908163
}
```

**Comment by KingJ** [comment]
Looking through the supported devices page, i've spotted the [ID-UK21FW09 rotary dimmer](https://www.zigbee2mqtt.io/devices/ID-UK21FW09.html) which looks identical. Indeed, if I follow the link through to the vendor's page the wiring diagram image is identical to that which is used in Candeo's documentation. The pictures of the rear are also identical aside from the change in vendor name.

I've copied the `configure` block from [the device definition for the ID-UK21FW09](url) and put it in to my customisation for the `HK-DIM-A` - although it is identical to the [Eco-Dim.07 definition](https://github.com/Koenkk/zigbee-herdsman-converters/blob/c1f440e3a63948266290eaacf6da779c026b1806/devices/ecodim.js#L41-L52) which my existing C202 is detected as. However, this still did not fix the issue - I could set the light on and off, but brightness could only be reported upon and not set.

Up until now, I was doing all of this testing entirely via zigbee2mqtt's front end. However, as a test I tried setting it via Home Assistant... which worked. I also then tried setting the brightness of my existing C202 (i.e. the one identified as an Eco-Dim.07) via zigbee2mqtt which also did not work (!). I think this comes down to an issue in the frontend, rather than with the device definitions, as;

- ✔️ Setting brightness via Home Assistant/MQTT
- ✔️ Setting brightness by dragging the slider in zigbee2mqtt's frontend on PC
- ✔️ Setting brightness by entering a number in zigbee2mqtt's frontend on mobile
- ❌ Setting brightness by dragging the slider in zigbee2mqtt's frontend on mobile

When I set up my first C202 I was doing it entirely via my PC since the light was close enough to see. This new dimmer however is a bit further away so i've been using my phone instead. So although a device definition needs adding/adjusting for this revision of the C202, the issues i've had with setting the brightness are seemingly down to the brightness slider within zigbee2mqtt itself not working on mobile.

What would be the best approach to add support for this device? Extend the [whiteLabel configuration of the ID-UK21FW09](https://github.com/Koenkk/zigbee-herdsman-converters/blob/c1f440e3a63948266290eaacf6da779c026b1806/devices/iolloi.js#L11), or add it as an entirely separate device? If the former, should I also extend that for my existing C202? The pictures, model ID and manufacturer ID does seem to have more in common with the ID-UK21FW09 than the Eco-Dim.07 (Pulling a few bits from the genBasic cluster, existing C202: `Read result of 'genBasic': {"manufacturerName":"Candeo","modelId":"Dimmer-Switch-ZB3.0","hwVersion":1}`, new C202: `{"manufacturerName":"Candeo","modelId":"HK-DIM-A","hwVersion":1}`).
